### PR TITLE
Move units logic to readascii support

### PR DESCRIPTION
**Pull request by Tom-Willemsen**
https://github.com/ISISComputingGroup/IBEX/issues/2279

### CODE REVIEW

File: ReadASCIIApp/src/get_metadata_impl.cpp [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    if (all_lines.size() == 0) {
+        // No commented block in file, return default.
+        return property_default;
+    }
+    
+    std::string header_line = all_lines.front();
+    all_lines.pop_front();
+    
+    if (header_line.find(file_format) == std::string::npos) {
+        // Header didn't contain our expected magic bytes - refuse to parse (might not be our format).
```
Comment (John-Holt-Tessella) [inline ReadASCIIApp/src/get_metadata_impl.cpp:73]: I think we should have an error here so people know why units are not set.

File: ReadASCIIApp/src/get_metadata_impl.cpp [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        // Commented block and magic bytes existed, but no other lines.
+        std::cerr << "get_metadata: magic bytes existed but no JSON." << std::endl;
+        return property_default;
+    }
+    
+    std::string all_lines_s = "";
+    for (auto const& value: all_lines) {
+        all_lines_s += value;
+    }
+
```
Comment (John-Holt-Tessella) [inline ReadASCIIApp/src/get_metadata_impl.cpp:87]: Can we check the format and warn if it is greater than 1.

File: ReadASCIIApp/src/get_metadata_impl.cpp [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    all_lines.pop_front();
+    
+    if (header_line.find(file_format) == std::string::npos) {
+        // Header didn't contain our expected magic bytes - refuse to parse (might not be our format).
+        return property_default;
+    }
+    
+    if (all_lines.size() == 0) {
+        // Commented block and magic bytes existed, but no other lines.
+        std::cerr << "get_metadata: magic bytes existed but no JSON." << std::endl;
```
Comment (John-Holt-Tessella) [inline ReadASCIIApp/src/get_metadata_impl.cpp]: Can we make the errors available as a PV eg outvb?

Comment (John-Holt-Tessella) [reply ReadASCIIApp/src/get_metadata_impl.cpp]: use the epic log error function here. but I think putting it in a pv would be good.

Comment (Tom-Willemsen) [reply ReadASCIIApp/src/get_metadata_impl.cpp]: changed to use epics errlog

File: ReadASCIIApp/src/get_metadata_impl.cpp (hunk 2 above)
Comment (Tom-Willemsen) [reply ReadASCIIApp/src/get_metadata_impl.cpp:87]: 👍

File: ReadASCIIApp/src/get_metadata_impl.cpp (hunk 1 above)
Comment (Tom-Willemsen) [reply ReadASCIIApp/src/get_metadata_impl.cpp:73]: Error is now in epics error log

**Pull request merged**